app/src/services/AdService.ts

```diff
@@ -1,23 +1,38 @@
 /* eslint-disable no-console */
-import { addDoc, collection, FirestoreError } from 'firebase/firestore';
+import { Book } from 'cremona';
+import { User } from 'firebase/auth';
+import {
+  collection,
+  FirestoreError,
+  getDocs,
+  query,
+  addDoc,
+  where,
+  doc,
+  deleteDoc,
+  updateDoc,
+} from 'firebase/firestore';
 import db from '../firebase/db';
 import ServiceSuccessResponse from './ServiceSuccessResponse';
 
 export interface Ad {
-  adDescription: string;
-  adPrice: number;
-  authors: string[];
-  bookDescription: string;
-  bookName: string;
-  courseCodes: string[];
-  tags: string[];
-  publishDate: Date;
-  wear: number;
-  isbn: number;
+  uid: string;
+  userId: string;
+  bookId: string;
+  price: number;
+  condition: string;
+  conditionDescribtion: string;
 }
 
+/**
+ * Handles all fetching and publishing of ads
+ *
+ * @version 1.0.0
+ * @author [Daniel Rygaard](https://github.com/Danryg)
+ * @author [Johan Blickhammar](https://github.com/JohanBlickhammar)
+ */
 export default class AdService {
-  static async addAd(ad: Ad): Promise<ServiceSuccessResponse> {
```

Comment:
I don't think you should need to specify and UID to publish an ad

Reply:
Agreed